### PR TITLE
styx environment and logging format

### PR DESCRIPTION
**Pull request by honnix**
@danielnorberg @ulzha PTAL

As discussed, it would be better to have a dedicated env.

### CODE REVIEW

**Comment by danielnorberg** [comment]
I think I want the opposite. Add a env var like `STYX_LOGGING=text` to quick-styx instead and look for that in our loggers.

**Comment by honnix** [comment]
This is not only for logging. We have use cases that users want to detect whether running on Styx. So we can give them a dedicated variable.

**Comment by ulzha** [comment]
Yeah I also think explicit STYX_LOGGING will be more obvious to work with down the line. I don't see a big benefit of immediately tying the choice of logging format to the fact of running in production Styx; the users may well want to be able to get per-task logging locally/in qstyx, and they should not have to specify a lying environment variable about running in production Styx to achieve that.

It does make sense to provide the injection of "running on Styx" information, separately. In general I'd say that a boolean toggle screams of a missed opportunity for extensibility; how about STYX_ENVIRONMENT that can be one of {production,testing,qstyx,...}?

**Comment by danielnorberg** [comment]
👍 for `STYX_LOGGING` and `STYX_ENVIRONMENT`

**Comment by honnix** [comment]
@danielnorberg @ulzha PTAL again.

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java [hunk 1]
```diff
@@ -681,11 +683,17 @@ private static DockerRunner createDockerRunner(
       LOG.info("Creating LocalDockerRunner");
       return closer.register(DockerRunner.local(scheduler, stateManager));
     } else {
+      final String styxEnvironment;
+      if (!config.hasPath(STYX_ENVIRONMENT)) {
+        styxEnvironment = STYX_ENVIRONMENT_PRODUCTION;
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: is production a good/safe default?

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: No idea. ;) Maybe `development` or `NA`?

**Comment by NatashaL** [comment]
@honnix is there more to be done here? can we merge? @ulzha any more feedback from you?

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java [hunk 2]
```diff
@@ -681,11 +683,17 @@ private static DockerRunner createDockerRunner(
       LOG.info("Creating LocalDockerRunner");
       return closer.register(DockerRunner.local(scheduler, stateManager));
     } else {
+      final String styxEnvironment;
+      if (!config.hasPath(STYX_ENVIRONMENT)) {
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: Would it be better perhaps to not have a default environment? I.e. fail if `styx.environment` is not set.

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: Afact this is how we handle `styx.mode`.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: Is it? I thought we just created different classes depending on `styx.mode`.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: Ah ok it fails by getting non-configured value.

**Comment by ulzha** [comment]
:+1:

**Comment by danielnorberg** [comment]
:+1:

**Pull request merged**